### PR TITLE
Cart functionality

### PR DESCRIPTION
**Pull request by muskaanshah**
Functionalities added -
Add to cart, delete from cart, delete from modal
Fix - drawer mobile responsiveness

Please note: Wishlist functionality is not yet added

### CODE REVIEW

File: src/context/cart-wishlist-context.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    switch (action.type) {
+        case "UPDATE_DEFAULT":
+            return { ...state, default: action.payload.value };
+        case "ADD_TO_CART":
+            return addToCart(state, action.payload.value);
+        case "REMOVE_FROM_CART":
+            return removeFromCart(state, action.payload.value, action.payload.isDeleteItem);
+        case "OPEN_MODAL":
+            const temp = { ...state, closeButton: !state.closeButton, idOfProduct: action.payload.value }
+            console.log(temp);
```
Comment (shobit1337) [inline src/context/cart-wishlist-context.js]: Remove the unnecessary console.logs

File: src/pages/home/Home.jsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ function Home() {
 				{categories.map((category) => {
 					return (
 						<Link to="/products">
```
Comment (shobit1337) [inline src/pages/home/Home.jsx]: Add a key attribute here.

**Review by shobit1337 — APPROVED** [review]
Looks good to merge.

File: src/pages/products/components/ProductCard.jsx [hunk 3]
```diff
@@ -10,21 +13,28 @@ function ProductCard({
 		badge,
 	},
 }) {
+	const { dispatch } = useCartWishlist();
 	let discount = Math.floor(((actualprice - newprice) * 100) / actualprice);
```
Comment (manismk) [inline src/pages/products/components/ProductCard.jsx]: can be converted to const. If it is not reassigned.

Suggestion: Also you can extract this into utils function as it has been used in product card and product card horizontal

**Review by Nikhil-Nirmal — APPROVED** [review]

**Comment by Nikhil-Nirmal** [comment]
Everything looks good

**Review by manismk — APPROVED** [review]
Looks good
Good naming convention and easy to understand

File: src/pages/products/components/ProductCard.jsx (hunk 3 above)
Comment (muskaanshah) [reply src/pages/products/components/ProductCard.jsx]: Will do, thank you!

**Pull request merged**